src/tardev-snapshotter/erofs-common/src/utils.rs

```diff
@@ -0,0 +1,187 @@
+use anyhow::{anyhow, Context, Result};
+use log::debug;
+use std::path::Path;
+use std::process::Command;
+use std::fs::{File, OpenOptions};
+use std::io::Write;
+use crate::constants::EROFS_METADATA_UUID;
+use crate::constants::EROFS_BLOCK_ALIGNMENT;
+
+/// Validates the input file paths for erofs operations.
+///
+/// This function performs common validation checks on the input paths:
+/// 1. Checks if paths are empty
+/// 2. Validates that files exist
+///
+/// # Arguments
+///
+/// * `tar_path` - Path to the tar file
+/// * `erofs_metadata_path` - Path to the erofs file
+/// * `check_tar_exists` - Whether to check if tar file exists
+/// * `check_erofs_metadata_exists` - Whether to check if erofs metadata file exists
+///
+/// # Returns
+///
+/// * `Result<()>` - Success or error
+fn validate_input_paths(
+    tar_path: &Path,
+    erofs_metadata_path: &Path,
+    check_tar_exists: bool,
+    check_erofs_metadata_exists: bool,
+) -> Result<()> {
+    // Check if input parameters are empty
+    if tar_path.as_os_str().is_empty() {
+        return Err(anyhow!("Tar path is empty"));
+    }
+
+    if erofs_metadata_path.as_os_str().is_empty() {
+        return Err(anyhow!("Erofs path is empty"));
+    }
+
+    // Check if tar file exists when required
+    if check_tar_exists && !tar_path.exists() {
+        return Err(anyhow!("Tar file does not exist: {:?}", tar_path));
+    }
+
+    // Check if erofs metadata file exists when required
+    if check_erofs_metadata_exists && !erofs_metadata_path.exists() {
+        return Err(anyhow!("Erofs metadata file does not exist: {:?}", erofs_metadata_path));
+    }
+
+    Ok(())
+}
+
+/// Creates an erofs metadata file from a decompressed tar file.
+///
+/// This function:
+/// 1. Takes a decompressed tar file path and output erofs metadata file path
+/// 2. Executes the mkfs.erofs command with appropriate options
+/// 3. Returns a Result indicating success or failure
+///
+/// # Arguments
+///
+/// * `decompressed_tar_path` - Path pointing to the decompressed tar file to process
+/// * `erofs_metadata_path` - Path where the output erofs metadata file will be created
+///
+/// # Returns
+///
+/// * `Result<()>` - Success or error
+
+pub fn create_erofs_metadata(
+    decompressed_tar_path: &Path, 
+    erofs_metadata_path: &Path,
+) -> Result<()> {
+    debug!(
+        "Creating erofs metadata {:?} from {:?}",
+        erofs_metadata_path, decompressed_tar_path
+    );
+
+    // Validate input paths - check tar exists but don't require erofs to exist yet
+    validate_input_paths(decompressed_tar_path, erofs_metadata_path, true, false)?;
```

Comment:
Do you not want to check explicitly that erofs_metadata_path file does not exist?

Reply:
At this point, the erofs_metadata_path file does not exist because we have not created it yet and so I did not want to check for it here. I check that it exists after we run the mkfs.erofs command at the bottom of `create_erofs_metadata`

Reply:
My point it we should check it does _not_ exist.

Reply:
Ohh I see. What should we do if it does exist though?

Reply:
Yup makes sense, updated the error checks here